LinkedList/Exercise/SLList.h

```diff
@@ -0,0 +1,328 @@
+#pragma once
+
+template<typename Type>
+class TSLList
+{
+private:
+
+	struct FNode
+	{
+		Type Element;
+		FNode* Next = nullptr;
+	};
+
+	FNode* Head;
+	FNode* Tail;
+
+	int ListSize;
+
+	FNode* GetNode(const int Index)
+	{
+		FNode* Current = Head;
+		for (int i = 0; i < Index; i++)
+		{
+			Current = Current->Next;
+		}
+		return Current;
+	}
+
+public:
+
+	class FIterator
+	{
+	private:
+		FNode* Current;
+
+	public:
+
+		FIterator(FNode* NewCurrent)
+		{
+			Current = NewCurrent;
+		}
+
+		FIterator& operator++ ()
+		{
+			Current = Current->Next;
+			return *this;
+		}
+
+		Type& operator* ()
+		{
+			return Current->Element;
+		}
+
+		bool operator!= (const FIterator& Other)
+		{
+			return Current != Other.Current;
+		}
+
+
+	};
+
+	FIterator begin()
+	{
+		if (Head)
+		{
+			return FIterator(Head);
+		}
+		return FIterator(nullptr);
+	}
+
+	FIterator end()
+	{
+		return FIterator(nullptr);
+	}
+
+
+	//Constructor
+	//-------------------------------------------------------------------------------------
+
+	TSLList() : Head(nullptr), Tail(nullptr), ListSize(0) {};
+
+	//-------------------------------------------------------------------------------------
+
+	~TSLList()
+	{
+		Clear();
+	};
+
+	//-------------------------------------------------------------------------------------
+
+	TSLList(TSLList& InList)
+	{
+		FNode* Current = InList.Head;
+		for (int i = 0; i < InList.ListSize; i++)
+		{
+			AddTail(Current->Element);
+			Current = Current->Next;
+		}
+	}
+
+	//-------------------------------------------------------------------------------------
+
+	Type& operator[] (const int Index)
+	{
+		return GetNode(Index)->Element;
+	}
+
+	//-------------------------------------------------------------------------------------
+
+	void operator= (TSLList& InList)
+	{
+		Clear();
+		FNode* Current = InList.Head;
+		for (int i = 0; i < InList.ListSize; i++)
+		{
+			AddTail(Current->Element);
+			Current = Current->Next;
+		}
+	}
+
+	//-------------------------------------------------------------------------------------
+
+	const Type& operator= (const int Index) const
+	{
+		return GetNode(Index)->Element;
+	}
+
+	//-------------------------------------------------------------------------------------
+
+	Type& GetHead()
+	{
+		return Head->Element;
+	}
+
```

Comment:
Perhaps add the //------------------------------------------------------------------------------------- division between these two